src/hooks/useNavigatePage.ts

```diff
@@ -0,0 +1,172 @@
+  const order = useMemo(
+    () => orderWithHidden?.filter((page) => !hiddenPages.has(page)),
+    [orderWithHidden, hiddenPages],
+  );
+
+  const currentPageId = pageKey ?? '';
+  const currentPageIndex = order?.indexOf(currentPageId) ?? -1;
+  const nextPageIndex = currentPageIndex !== -1 ? currentPageIndex + 1 : -1;
+  const previousPageIndex = currentPageIndex !== -1 ? currentPageIndex - 1 : -1;
+
+  /**
+   * For stateless apps, this is how we redirect to the
+   * initial page of the app. We replace the url, to not
+   * have the initial page (i.e. the page without a
+   * pageKey) in the history.
+   */
+  useEffect(() => {
+    if (isStatelessApp && order?.[0] !== undefined && !currentPageId) {
+      navigate(`/${order?.[0]}`, { replace: true });
+    }
+  }, [isStatelessApp, order, navigate, currentPageId]);
+
+  const navigateToPage = useCallback(
+    (page?: string, options?: NavigateToPageOptions) => {
+      if (!page) {
+        return;
+      }
+      setFocusId(options?.focusComponentId);
+      if (options?.returnToView) {
+        setReturnToView(options.returnToView);
+      }
+
+      if (autoSaveBehavior === 'onChangePage' && order?.includes(currentPageId)) {
+        dispatch(FormDataActions.saveLatest({}));
+      }
+
+      if (isStatelessApp) {
+        return navigate(`/${page}`);
+      }
+
+      const url = `/instance/${partyId}/${instanceGuid}/${taskId}/${page}`;
+      navigate(url);
+    },
+    [
+      navigate,
+      partyId,
+      instanceGuid,
+      taskId,
+      setFocusId,
+      setReturnToView,
+      autoSaveBehavior,
+      dispatch,
+      order,
+      currentPageId,
+      isStatelessApp,
+    ],
+  );
+
+  const isValidPageId = useCallback(
+    (pageId: string) => {
+      if (taskType !== ProcessTaskType.Data && ['confirmation', 'receipt', 'feedback'].includes(pageId)) {
+        return true;
+      }
+      if (taskType === ProcessTaskType.Confirm && pageId === 'confirmation') {
+        return true;
+      }
+      if (taskType === ProcessTaskType.Archived && pageId === 'receipt') {
+        return true;
+      }
+      if (taskType === ProcessTaskType.Feedback && pageId === 'feedback') {
+        return true;
+      }
+      return order?.includes(pageId) ?? false;
+    },
+    [order, taskType],
+  );
+
+  const navigateToTask = useCallback(
+    (taskId?: string, options?: NavigateOptions) => {
+      const url = `/instance/${partyId}/${instanceGuid}/${taskId ?? lastTaskId}`;
+      navigate(url, options);
+    },
+    [partyId, instanceGuid, lastTaskId, navigate],
+  );
+
+  const isCurrentTask = useMemo(() => currentTaskId === taskId, [currentTaskId, taskId]);
+
+  const startUrl = useMemo(() => {
+    if (taskType === ProcessTaskType.Confirm) {
+      return `/instance/${partyId}/${instanceGuid}/${taskId}/confirmation`;
+    }
+    if (taskType === ProcessTaskType.Archived) {
+      return `/instance/${partyId}/${instanceGuid}/ProcessEnd/receipt`;
```

Comment:
The string `ProcessEnd` could probably also have been added to an enum or been made into a global const. I've seen it at least twice now, so there's magic tied to it, but it would be nice not having to search around for that string to find the magic.